lib/ranked-model/ranker.rb

```diff
@@ -69,6 +70,19 @@ def update_rank! value
           update_all(ranker.column => value)
       end
 
+      def insert_after(previous_ranker)
+        if previous_ranker.instance.class != instance.class
+          raise ObjectMismatchError, 'The previous object sent does not belong to the same model as self'
+        end
+
+        preceeding_object_rank = previous_ranker.relative_rank
+
+        new_rank = preceeding_object_rank > relative_rank ? preceeding_object_rank : preceeding_object_rank + 1
```

Comment:
I think this may not be necessary. I think the reordering code already copes with the object being moved  to either side of a relative position value.

Reply:
So, this is necessary because the reordering code will work depending on what rank we send. Now, this line here sets the correct rank.

Ex - In the following list - ItemA, ItemB, ItemC, ItemD - If ItemA needs to be inserted after ItemC, then the final list is - ItemB, ItemC, ItemA, ItemD.

We can see that ItemA's new rank should be equal to 2.
If we just do ItemC's rank +1 - we would have got the new rank as `ItemC's rank + 1 => 3` which would then place ItemA at the end of the list.
In this particular scenario, we need to ignore ItemA from the list when calculating the position of ItemC. So the new position effectively is the same as ItemC's original rank.

Reply:
I see what you mean :)